inc/Engine/License/views/upgrade-popin.php

```diff
@@ -0,0 +1,48 @@
+<?php
+/**
+ * Upgrade section template.
+ *
+ * @since 3.7.3
+ */
+
+defined( 'ABSPATH' ) || exit;
+?>
+<div class="wpr-Popin wpr-Popin-Upgrade">
+	<div class="wpr-Popin-header">
+		<h2 class="wpr-title1"><?php esc_html_e( 'Speed Up More Websites', 'rocket' ); ?></h2>
+		<button class="wpr-Popin-close wpr-Popin-Upgrade-close wpr-icon-close"></button>
+	</div>
+	<div class="wpr-Popin-content">
+		<p>
+		<?php
+		// translators: %1$s = opening strong tag, %2$s = closing strong tag.
+		printf( esc_html__( 'You can use WP Rocket on more websites by upgrading your license. To upgrade, simply pay the %1$sprice difference%2$s between your current and new licenses, as shown below.', 'rocket' ), '<strong>', '</strong>' );
+		?>
+		</p>
+		<p>
+		<?php
+		// translators: %1$s = opening strong tag, %2$s = closing strong tag.
+		printf( esc_html__( '%1$sN.B.%2$s: Upgrading your license does not change your expiration date', 'rocket' ), '<strong>', '</strong>' );
+		?>
+		</p>
+		<div class="wpr-Popin-flex">
+			<?php foreach ( $data['upgrades'] as $rocket_upgrade ) : ?>
+			<div>
+				<h3><?php echo esc_html( $rocket_upgrade['name'] ); ?></h3>
```

Comment:
@Tabrisrp Is the array structure correct here? Why? 

In method `WP_Rocket\Engine\License\Upgrade\get_upgrade_choices()`, the first dimension is the license type, ie `plus`, `infinite`. And then the `name` and `price` are within the license type. Am I reading this right?

Reply:
the `plus` and `infinite` in `get_upgrade_choices()` are named indexes for the array, not a first dimension.

So the structure is correct, each array within the `upgrades` are directly with the corresponding upgrade data.

Reply:
Are they though? Maybe I'm missing something. So let me walkthrough my understanding and see if it makes sense.

For example, let's look at this code:

```
		if ( $license === $this->pricing->get_single_websites_count() ) {
			$choices['plus']     = $this->get_upgrade_from_single_to_plus_data();
			$choices['infinite'] = $this->get_upgrade_from_single_to_infinite_data();
```
after running this, the array in `choices` would be:

```php
'plus' => [
			'name'        => 'Plus',
			'price'       => 49,
			'websites'    => 1,
			'upgrade_url' => 'https://.....',
],
'infinite' => [
			'name'        => 'Infinite',
			'price'       => 150,
			'websites'    => 1,
			'upgrade_url' => 'https://.....',
]
```

and then the data is populated here:

```php
		$data = [
			'upgrades' => $this->get_upgrade_choices(),
		];
```
to where `$data['upgrades']` would have the above array structure.

What am I missing?

Reply:
It's here at this moment where I'm saying to myself: Doh, Tonya, Doh.